sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -111,16 +111,17 @@ def test_classification_toy(loss):
         ({"max_depth": -0.1}, "max_depth must be greater than zero"),
         ({"max_depth": 0}, "max_depth must be greater than zero"),
         ({"init": {}}, "The init parameter must be an estimator or 'zero'"),
-        ({"max_features": "invalid"}, "Invalid value for max_features:"),
-        ({"max_features": 0}, r"max_features must be in \(0, n_features\]"),
-        ({"max_features": 100}, r"max_features must be in \(0, n_features\]"),
-        ({"max_features": -0.1}, r"max_features must be in \(0, n_features\]"),
+        ({"max_features": "invalid"}, ValueError, "Invalid value for max_features:"),
+        ({"max_features": -0.1}, ValueError, "max_features == -0.1, must be > 0."),
+        ({"max_features": 0}, ValueError, "max_features == 0, must be > 0."),
+        ({"max_features": 100}, ValueError, "max_features == 100, must be <= "),
```

Comment:
This test is only passing right now because I didn't include a concrete number in the error message to represent `max_val`(which is set to `n_features_in_`). The number should be `4` for gbr (tested with `X_reg`) and `10` for gbc (tested with`iris.data`).

Is there is a suggestion for how to better generalize this `max_val` in the parameterized testing?